### DISCUSSION TITLE
Prevent lockout when forgetting to provide ssh key for rule 5.1.24

### DISCUSSION THREAD

**Discussion by Luiggi33** [MVladislav/ansible-cis-ubuntu-2404#6] [General]
I locked myself out of my vm on accident, after running the example playbook and forgetting to put the pubkey in it. Would it be practical to keep the "status quo", when a rescue block for said rule is triggered?

**Comment by MVladislav** [comment]
I don't think this is the first time I’ve heard that.

What I could do is change the process in the following way:

- The rule `cis_ubuntu2404_rule_5_1_23` is an extended one, not defined in CIS. I will disable this by default.
- If `cis_ubuntu2404_rule_5_1_23` is enabled and `cis_ubuntu2404_ssh_password_authentication` is set to `no`, the rule will only be executed when `cis_ubuntu2404_rule_5_1_24_ssh_pub_key` is also defined.

The part where I'm currently unsure of the best approach is:

- Should the process stop entirely if `cis_ubuntu2404_rule_5_1_23` is active but `cis_ubuntu2404_rule_5_1_24_ssh_pub_key` is not set?
- Or should the execution of `cis_ubuntu2404_rule_5_1_23` simply be skipped if `cis_ubuntu2404_rule_5_1_24_ssh_pub_key` is not set?
  - With this approach, you wouldn’t know (unless checked) that `cis_ubuntu2404_rule_5_1_23` wasn't executed due to the absence of `cis_ubuntu2404_rule_5_1_24_ssh_pub_key`.

**Reply by Luiggi33** [reply]
I think the second option makes more sense. I don't think anyone would (without double checking the config and output) run this on their personal machines. From what I understand, these roles are mostly used when you want to deploy VMs or servers automatically. In these contexts, it should be okay to just skip it and *maybe* give a warning that it got skipped

**Reply by MVladislav** [reply]
Change is included in 19eb72e441946648b165b3f18cfa7782c8b09d7f
